### ISSUE TITLE
[Feature Request] Setting # train steps per epoch

### ISSUE THREAD

**Issue by kl2792** [pytorch/ignite#226]
In the case that a dataset is very large (>500k training samples) and you want to see validation loss periodically throughout the dataset.

Happy to put in a pull request implementing this -- any pointers on where to start?

**Comment by vfdev-5** [comment]
@kl2792 this can be done by setting the validation handler on `Events.ITERATION_COMPLETED`:
```
trainer = ...
evaluator = ...

val_interval = 10000

@trainer.on(Events.ITERATION_COMPLETED)
def log_validation_results(engine):
    if engine.state.iteration % val_interval == 0:
        evaluator.run(val_loader)
```

**Comment by alykhantejani** [comment]
Closing as this can be handled with the ITERATION_COMPLETED event as @vfdev-5 says. @kl2792 if this doesn't fully answer please feel free to comment back on this issue

**Comment by vfdev-5** [comment]
Here is an example implementation what I'm talking about:
```python
from enum import Enum
from ignite.engine import Engine, Events

loss_values = list(range(500, 0, -1))
loss_iter = iter(loss_values)

def update(*args, **kwargs):
    return next(loss_iter)


engine = Engine(update)

### THE FOLLOWING CAN BE WRAPPED IN A SPECIAL HANDLER ####

class CustomEpochEvents(Enum):
    CUSTOM_EPOCH_STARTED = "custom_epoch_started"
    CUSTOM_EPOCH_COMPLETED = "custom_epoch_completed"

engine.register_events(*CustomEpochEvents)
custom_epoch_length = 3

@engine.on(Events.STARTED)
def on_started(engine):
    engine.state.custom_epoch = 0


@engine.on(Events.ITERATION_STARTED)
def on_custom_epoch_started(engine):
    if engine.state.iteration % custom_epoch_length == 1:
        engine.state.custom_epoch += 1
        engine.fire_event(CustomEpochEvents.CUSTOM_EPOCH_STARTED)


@engine.on(Events.ITERATION_COMPLETED)
def on_custom_epoch_started(engine):
    if engine.state.iteration % custom_epoch_length == 0:
        engine.fire_event(CustomEpochEvents.CUSTOM_EPOCH_COMPLETED)

### end of THE FOLLOWING CAN BE WRAPPED IN A SPECIAL HANDLER ####        

data = list(range(16))

@engine.on(Events.ITERATION_STARTED)
def on_iteration_started(engine):
    print("- Iteration {} started, epoch {}".format(engine.state.iteration, engine.state.epoch))

@engine.on(Events.ITERATION_STARTED)
def on_iteration_ended(engine):
    print("-- Iteration {} ended, epoch {}".format(engine.state.iteration, engine.state.epoch))
    
    
@engine.on(Events.EPOCH_STARTED)
def on_epoch_started(engine):
    print("Epoch {} started".format(engine.state.epoch))
    

# user handlers
@engine.on(CustomEpochEvents.CUSTOM_EPOCH_STARTED)
def on_my_epoch_started(engine):
    print("Custom epoch {} started at iteration {}".format(engine.state.custom_epoch, engine.state.iteration))


@engine.on(CustomEpochEvents.CUSTOM_EPOCH_COMPLETED)
def on_my_epoch_started(engine):
    print("Custom epoch {} ended at iteration {}".format(engine.state.custom_epoch, engine.state.iteration))

engine.run(data, max_epochs=2)
```
what gives:
```
Epoch 1 started
Custom epoch 1 started at iteration 1
- Iteration 1 started, epoch 1
-- Iteration 1 ended, epoch 1
- Iteration 2 started, epoch 1
-- Iteration 2 ended, epoch 1
- Iteration 3 started, epoch 1
-- Iteration 3 ended, epoch 1
Custom epoch 1 ended at iteration 3
Custom epoch 2 started at iteration 4
- Iteration 4 started, epoch 1
-- Iteration 4 ended, epoch 1
- Iteration 5 started, epoch 1
-- Iteration 5 ended, epoch 1
- Iteration 6 started, epoch 1
-- Iteration 6 ended, epoch 1
Custom epoch 2 ended at iteration 6
Custom epoch 3 started at iteration 7
- Iteration 7 started, epoch 1
-- Iteration 7 ended, epoch 1
- Iteration 8 started, epoch 1
-- Iteration 8 ended, epoch 1
- Iteration 9 started, epoch 1
-- Iteration 9 ended, epoch 1
Custom epoch 3 ended at iteration 9
Custom epoch 4 started at iteration 10
- Iteration 10 started, epoch 1
-- Iteration 10 ended, epoch 1
- Iteration 11 started, epoch 1
-- Iteration 11 ended, epoch 1
- Iteration 12 started, epoch 1
-- Iteration 12 ended, epoch 1
Custom epoch 4 ended at iteration 12
Custom epoch 5 started at iteration 13
- Iteration 13 started, epoch 1
-- Iteration 13 ended, epoch 1
- Iteration 14 started, epoch 1
-- Iteration 14 ended, epoch 1
- Iteration 15 started, epoch 1
-- Iteration 15 ended, epoch 1
Custom epoch 5 ended at iteration 15
Custom epoch 6 started at iteration 16
- Iteration 16 started, epoch 1
-- Iteration 16 ended, epoch 1
Epoch 2 started
- Iteration 17 started, epoch 2
-- Iteration 17 ended, epoch 2
...
```
